### DISCUSSION TITLE
Neovim version manager

### DISCUSSION THREAD

**Discussion by Himujjal** [abenz1267/neoconf#4] [Ideas]
A simple version manager for neovim distribution should be added. Similar to rustup for rust and nvm for node.

**Why this feature?**

This will make neoconf the default tool to install instead of managing dependencies manually in different OS.

**Difficulty level?**

With git installed on the machine this shouldn't be that difficult to build anyways.

**Comment by abenz1267** [comment]
What would a possible scenario be, though? Why would you want to switch neovim versions? I assume for switching between stable and nightly? Would the OS have both versions installed? Which version should neoconf work with? What about plugin version-mismatches?

I think while it sounds easy on the surface, there are a lot of considerations to be made first.

**Reply by Himujjal** [reply]
The possible scenario is of course switching between stable and nightly.

This is beneficial to devs working on neovim core mostly. Like all nightlies.

Anyways, this can be an optional goal.

**Comment by abenz1267** [comment]
So i've been tinkering around with the idea over the last few days and i think i got a solution that'd move this kinda functionality into the right direction: i will introduce **neoconf environments**.

This will go beyond just choosing the neovim version, as that itself comes with other possibly problems, as i mentioned before.

## So what will a **neoconf environment** be?

It's basically just a locally saved complete configuration you can swap via `neoconf switch`. This way you can have completely different configurations for different environments. You'd also be able to define which neovim version to use: nightly or latest stable. Updating of those binaries will be managed by `neoconf update`, as the general update process.

Regards

**Reply by Himujjal** [reply]
Yeah. This is what I suggested. Let me go through the code though.
